packages/block-editor/src/components/publish-date-time-picker/index.js

```diff
@@ -17,10 +17,18 @@ export function PublishDateTimePicker(
 		onChange,
 		showPopoverHeaderActions,
 		isCompact,
+		currentDate,
 		...additionalProps
 	},
 	ref
 ) {
+	const datePickerProps = {
+		startOfWeek: getSettings().l10n.startOfWeek,
+		onChange,
+		currentDate: isCompact ? undefined : currentDate,
+		currentTime: isCompact ? currentDate : undefined,
```

Comment:
It's not clear to me why isCompact impacts currentData prop? 

Do we want the date to be "controlled" outside the entity in the case of the post status. What's happening exactly here?

Reply:
TimePicker component has [different name ](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/date-time/time/index.tsx#L96)for the `currentDate` and is `currentTime`.

Reply:
ah ok then this fix makes sense :P 

Both of these components should just use value onChange to be honest.